lib/filewatch/watch.rb

```diff
@@ -62,62 +45,66 @@ def initialize(opts={})
       end
       @watching = []
       @exclude = []
-      @files = Hash.new { |h, k| h[k] = WatchedFile.new(k, false, false) }
-      @unwatched = Hash.new
+      @files = Hash.new
       # we need to be threadsafe about the mutation
       # of the above 2 ivars because the public
-      # methods each, discover, watch and unwatch
+      # methods each, discover and watch
       # can be called from different threads.
       @lock = Mutex.new
       # we need to be threadsafe about the quit mutation
       @quit = false
       @quit_lock = Mutex.new
+      @max_active = ENV.fetch("FILEWATCH_MAX_OPEN_FILES", 4095).to_i
```

Comment:
My feeling is that this shouldn't be configurable from the environment. However, we can leave it as-is as long as we don't document it.